resource-aggregate/commander/commander.go

```diff
@@ -0,0 +1,266 @@
+package commander
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/plgd-dev/cloud/resource-aggregate/commands"
+	"github.com/plgd-dev/cloud/resource-aggregate/cqrs/eventbus"
+	"github.com/plgd-dev/cloud/resource-aggregate/cqrs/utils"
+	"github.com/plgd-dev/cloud/resource-aggregate/events"
+	"github.com/plgd-dev/cloud/resource-aggregate/service"
+)
+
+// Commander provides sync commands.
+type Commander struct {
+	subscriber eventbus.Subscriber
+	raClient   service.ResourceAggregateClient
+}
+
+// NewCommander instancies commander
+func NewCommander(subscriber eventbus.Subscriber, raClient service.ResourceAggregateClient) *Commander {
+	return &Commander{
+		subscriber: subscriber,
+		raClient:   raClient,
+	}
+}
+
+type deleteHandler struct {
+	correlationID string
+	res           chan *events.ResourceDeleted
+}
+
+func newDeleteHandler(correlationID string) *deleteHandler {
+	return &deleteHandler{
+		correlationID: correlationID,
+		res:           make(chan *events.ResourceDeleted, 1),
+	}
+}
+
+func (h *deleteHandler) Handle(ctx context.Context, iter eventbus.Iter) (err error) {
+	for {
+		ev, ok := iter.Next(ctx)
+		if !ok {
+			return iter.Err()
+		}
+		var s events.ResourceDeleted
+		if ev.EventType() == s.EventType() {
+			if err := ev.Unmarshal(&s); err != nil {
+				return err
+			}
+			if s.GetAuditContext().GetCorrelationId() == h.correlationID {
+				select {
+				case h.res <- &s:
+				default:
+				}
+				return nil
+			}
+		}
+	}
+}
+
+func (h *deleteHandler) recv(ctx context.Context) (*events.ResourceDeleted, error) {
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case v := <-h.res:
+		return v, nil
+	}
+}
+
+type updateHandler struct {
+	correlationID string
+	res           chan *events.ResourceUpdated
+}
+
+func newUpdateHandler(correlationID string) *updateHandler {
+	return &updateHandler{
+		correlationID: correlationID,
+		res:           make(chan *events.ResourceUpdated, 1),
+	}
+}
+
+func (h *updateHandler) Handle(ctx context.Context, iter eventbus.Iter) (err error) {
```

Comment:
Is this supposed to be just a helper so each gateway is not implementing it on it's own? Why it's not containing all commands RA support?

Reply:
Only these commands waits for events from the target device.